ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1031,6 +1031,35 @@ extension SignalType {
 	}
 }
 
+extension SignalType where Value: Hashable {
+	/// Forwards only those values from `self` that are unique across the set of
+	/// all values that have been seen.
+	/// Note: This causes values that are forwarded to be retained to check for
+	/// uniquness.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func uniqueValues() -> Signal<Value, Error> {
+		return Signal { observer in
+			let seenValues: Atomic<Set<Value>> = Atomic([])
+			
+			return self
+				.filter { value in return seenValues.withValue { !$0.contains(value) } }
+				.observe { event in
+					switch event {
+					case let .Next(value):
+						seenValues.modify { set in
+							var mutableSet = set
+							mutableSet.insert(value)
+							return mutableSet
+						}
```

Comment:
I think there's a race condition here between the `filter` and `observe`. Can't a repeat of a value could make it past the `filter` before the first occurrence makes it into `observe`?

I think we need to filter and modify the seen values in a single atomic operation.